internal/api/backresthandler.go

```diff
@@ -886,3 +913,82 @@ func (s *BackrestHandler) GetSummaryDashboard(ctx context.Context, req *connect.
 
 	return connect.NewResponse(response), nil
 }
+
+func humanTime(d time.Duration) string {
+	d = d.Round(time.Second)
+	days := d / (24 * time.Hour)
+	d -= days * 24 * time.Hour
+	h := d / time.Hour
+	d -= h * time.Hour
+	m := d / time.Minute
+	d -= m * time.Minute
+	s := d / time.Second
+	if days > 0 {
+		return fmt.Sprintf("%dd %02dh:%02dm:%02ds", days, h, m, s)
+	}
+	return fmt.Sprintf("%02dh:%02dm:%02ds", h, m, s)
+}
+
+func formatSize(size int64) string {
+	if size < 1024 {
+		return fmt.Sprintf("%d B", size)
+	}
+	const unit = 1024
+	if size < unit*unit {
+		return fmt.Sprintf("%.2f KB", float64(size)/unit)
+	}
+	if size < unit*unit*unit {
+		return fmt.Sprintf("%.2f MB", float64(size)/(unit*unit))
+	}
+	if size < unit*unit*unit*unit {
+		return fmt.Sprintf("%.2f GB", float64(size)/(unit*unit*unit))
+	}
+	return fmt.Sprintf("%.2f TB", float64(size)/(unit*unit*unit*unit))
+}
+
+func directorySize(path string) string {
```

Comment:
Hmm, I'm a bit nervous about the performance of this function. In principle checking the size is a neat idea, in practice it's likely to be the slowest part of loading the status summary dashboard as it's implemented right now.

My inclination is to say it should be dropped, or if we report something it should be the sizes of specific stores e.g. # total operations, # total operation log bytes used, etc. Same goes for logs, but I'm not sure I want to expose info at that level of granularity.

Reply:
Adjusted to use the function